docs/api/implementation-guide.md

```diff
@@ -0,0 +1,100 @@
+---
+title: NuGet Server Implementation Guide
+description: Guidelines and recommendations to anyone implementing the NuGet Server API in their own package repository
+author: zivkan
+ms.author: zivkan
+ms.date: 07/29/2023
+ms.topic: conceptual
+---
+
+# NuGet Server Implementation Guide
+
+Prior to 2018, NuGet package repositories could copy the package metadata from the `nupkg`'s [`nuspec` file](../reference/nuspec.md), and return the results unmodified in [search](./search-query-service-resource.md) and [package metadata](./registration-base-url-resource.md) requests.
+Since then, NuGet has evolved to provide developers with a richer experience, and this requires package repositories to do additional work in order to provide the additional value to developers.
+
+## Chronology
```

Comment:
Maybe have a section which is "Required endpoints" where you simply link to this section: https://learn.microsoft.com/en-us/nuget/api/overview#resources-and-schema and tell the reader to refer to the "required" column.

Reply:
Other than semver2, nothing seems essential (and given how old NuGet 3.6 is, supporting semver1 only clients seems like a non-issue any longer). Client can't assume that just because nuget.org implemented something that other servers have as well, so I really find it hard to imagine any change in the future that is truly required. 🤷

Reply:
Sounds good.